ui/src/message_popup/author/question_page.jsx

```diff
@@ -0,0 +1,284 @@
+import {learningObjectives} from '../../data/learning_objectives.js';
+import {indicators} from '../../data/indicators.js';
+import * as Routes from '../../routes.js';
+
+import AppBar from 'material-ui/AppBar';
+import IconButton from 'material-ui/IconButton';
+import RaisedButton from 'material-ui/RaisedButton';
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+
+import ArrowBackIcon from 'material-ui/svg-icons/navigation/arrow-back';
+
+export default React.createClass({
+  displayName: 'QuestionPage',
+
+  propTypes: {
+    originalQuestion: React.PropTypes.object,
+    questionText: React.PropTypes.string,
+    students: React.PropTypes.array,
+    learningObjective: React.PropTypes.object,
+    indicator: React.PropTypes.object,
+    goodExamplesText: React.PropTypes.string,
+    badExamplesText: React.PropTypes.string
+  },
+
+  getDefaultProps(){
+    return ({
+      questionText: '',
+      students: [],
+      learningObjective: learningObjectives[0],
+      indicator: indicators[0],
+      goodExamplesText: "",
+      badExamplesText: ""
+    });
+
+  },
+  getInitialState() {
+    return ({
+      questionText: this.props.questionText,
+      students: this.props.students,
+      learningObjective: this.props.learningObjective,
+      indicator: this.props.indicator,
+      goodExamplesText: this.props.goodExamplesText,
+      badExamplesText: this.props.badExamplesText,
+      deleteConfirmationOpen: false
+    });
+  },
+
+  returnToQuestions(){
+    Routes.navigate(Routes.messagePopupAuthorQuestionsPath());
+  },
+
+  onQuestionTextChange(event, value){
+    this.setState({questionText: value});
+  },
+
+  addStudent(studentName){
+    const student = _.find(allStudents, student => student.name.toLowerCase() === studentName.toLowerCase());
+    const alreadyAdded =  student !== undefined && _.find(this.state.students, newStudent => newStudent.id === student.id) !== undefined;
+    if(student !== undefined && !alreadyAdded){
+      this.setState({students: this.state.students.concat(student)});
+    }
+  },
+
+  removeStudent(studentId){
+    return function(){
+      this.setState({students: _.remove(_.clone(this.state.students), student => student.id !== studentId)});
+    }.bind(this);
+  },
+
+  onGoodExamplesChange(event, text){
+    this.setState({goodExamplesText: text});
+    this.getExamples();
+  },
+
+  onBadExamplesChange(event, text){
+    this.setState({badExamplesText: text});
+    this.getExamples();
+  },
+
+  getExamples(){
+    const {goodExamplesText, badExamplesText} = this.state;
+    var goodExamples = goodExamplesText.split('\n\n');
+    var badExamples = badExamplesText.split('\n\n');
+    goodExamples = goodExamples.map(example => example.trim());
+    badExamples = badExamples.map(example => example.trim());
+    _.remove(goodExamples, example => example == '');
+    _.remove(badExamples, example => example == '');
```

Comment:
This seems like a candidate for a method that you could use for both good and bad examples, something like `parseExampleText` that takes a string and does this transformation?